src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,263 @@
+    // sys_id is mandatory for update operation
+    if (jsonElement == null && config.getOperation().equals(ServiceNowConstants.UPDATE_OPERATION)) {
+      throw new IllegalStateException("No sys_id found in the record to be updated");
+      // sys_id need to be removed for insert operation
+    } else if (jsonElement != null && config.getOperation().equals(ServiceNowConstants.INSERT_OPERATION)) {
+      jsonObject.remove(ServiceNowConstants.SYS_ID);
+      // this condition will run when jsonElement is not null and operation is update
+    } else if (jsonElement != null) {
+      sysId = jsonElement.getAsString();
+    }
+    String data = jsonObject.toString();
+    String encodedData = Base64.getEncoder().encodeToString(data.getBytes(StandardCharsets.UTF_8));
+
+    RestRequest restRequest = new RestRequest();
+    restRequest.setUrl(getUrl(sysId));
+    restRequest.setId(UUID.randomUUID().toString());
+    restRequest.setHeaders(getHeaders());
+    restRequest.setMethod(ServiceNowConstants.HTTP_POST);
+    if (config.getOperation().equals(ServiceNowConstants.UPDATE_OPERATION)) {
+      restRequest.setMethod(ServiceNowConstants.HTTP_PUT);
+    }
+    restRequest.setBody(encodedData);
+    return restRequest;
+  }
+
+  /**
+   * Inserts/Updates the list of records into ServiceNow table
+   *
+   * @param restRequestsMap The map of restRequests
+   * @param accessToken    The access token
+   */
+  public void createPostRequest(Map<String, RestRequest> restRequestsMap, String accessToken)
+    throws OAuthProblemException, InterruptedException {
+    ServiceNowBatchRequest payloadRequest = getPayloadRequest(restRequestsMap);
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint());
+    RestAPIResponse apiResponse;
+
+    try {
+      requestBuilder.setAuthHeader(accessToken);
+      requestBuilder.setAcceptHeader(MediaType.APPLICATION_JSON);
+      requestBuilder.setContentTypeHeader(MediaType.APPLICATION_JSON);
+      StringEntity stringEntity = new StringEntity(gson.toJson(payloadRequest), ContentType.APPLICATION_JSON);
+      requestBuilder.setEntity(stringEntity);
+      apiResponse = restApi.executePost(requestBuilder.build());
+
+      if (!apiResponse.isSuccess()) {
+        LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+        throw new RetryableException();
+      } else {
+        JsonObject responseJSON = jsonParser.parse(apiResponse.getResponseBody()).getAsJsonObject();
+        JsonArray servicedRequestsArray = responseJSON.get(ServiceNowConstants.SERVICED_REQUESTS).getAsJsonArray();
+        JsonElement failedRequestId = null;
+        for (int i = 0; i < servicedRequestsArray.size(); i++) {
+          int statusCode = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.STATUS_CODE)
+            .getAsInt();
+          if (statusCode == HttpStatus.SC_FORBIDDEN || statusCode == HttpStatus.SC_BAD_REQUEST ||
```

Comment:
Added a case when ServiceNow Table API returns HTTP status code: 404 if _sys_id_ present in the request payload is not found in ServiceNow. (For update operation)


The following error message is returned by the API in this case : 
```
{
  "error": {
    "message": "No Record found",
    "detail": "Record doesn't exist or ACL restricts the record retrieval"
  },
  "status": "failure"
}
```

141d14bc854fc0889d5a6fabae03e8127abe5c59

Reply:
Instead of checking specific error codes it would be better to handle all 4xx and 5xx codes as errors. So in the future, if the ServiceNow API returns a different error code like 503 (Service Unavailable), this code should still work as expected.
For example, see https://github.com/cdapio/cdap/blob/develop/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java#L870

Reply:
Refactored the **if** condition to check for all 4xx as well as 5xx HTTP Status Codes f453a61bee4a0cffcfd69509adb54fa1b7933f27